### PR TITLE
Implentation of absolute value function.

### LINKED ISSUE #16: Absolute Value
Objective:
I would like to work on the Absolute Value function.
This function calculates and returns the absolute value of a given number.

**Behavior Description:**

- The function should take a single numerical input (integer or float).
- If the input is positive or zero, it should return the input unchanged.
- If the input is negative, it should return the positive equivalent of the number.
- The function should handle edge cases like zero and very large/small numbers.
- Non-numeric inputs should raise an appropriate error.

### PR DESCRIPTION
**Pull request by abeddost**
---
name: solution review 
about: A template PR for code review with a checklist
---

# Behavior

## Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

Hi team, could someone please review this Pull Request?
Closes #16

### CODE REVIEW

File: solutions/absolute_value.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    >>> absolute_value(10)
+    10
+
+    >>> absolute_value(-10)
+    10
+    """
+    # the number should be an integer or float
+    if not isinstance(number, (int, float)):
+        raise TypeError("The input must be an int or float")
```
Comment (zeozeph) [inline solutions/absolute_value.py:40]: The TypeError message sounds too generic, how about tweaking it to not just be specific but also to be able to show that it identified the input and consequently, prompts specific responses. 
example: TypeError(f"Invalid entry: {number}. The input must be an int or a float")

File: solutions/tests/test_absolute_value.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        expected = 1000000
+        self.assertEqual(actual, expected)
+
+    def test_absolute_value_of_small_number(self):
+        """It should handle small numbers correctly"""
+        actual = absolute_value(0.000001)
+        expected = 0.000001
+        self.assertEqual(actual, expected)
+
+    def test_type_error_for_non_number_input(self):
```
[the remark this answers is not in the document]
Comment (abeddost) [reply solutions/tests/test_absolute_value.py:53]: Hi @zeozeph! The test ensures absolute_value("hello") raises a TypeError. Since assertRaises only checks for the exception, it doesn’t compare expected and actual values. It just confirms the right error happens.

File: solutions/tests/test_absolute_value.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        """It should raise a TypeError for non-number input"""
```
[the remark this answers is not in the document]
Comment (abeddost) [reply solutions/tests/test_absolute_value.py:54]: Thank you so much for your feedback! Assertions are used to check if something works as expected. If it doesn’t, an error is raised. In this case, we’re not looking for a specific value, just making sure the behavior is correct. That’s how Python handles assertions.

File: .vscode/settings.json
[the remark this answers is not in the document]
Comment (abeddost) [reply .vscode/settings.json:1]: Hi @zeozeph! Thank you for your feedback. I have made the required changes.

File: solutions/absolute_value.py (hunk 1 above)
Comment (abeddost) [reply solutions/absolute_value.py]: Hey @zeozeph! Thank you for your feedback. I made the changes; please have a look at them.

Comment (zeozeph) [reply solutions/absolute_value.py]: I have seen the changes you made, they really addressed the issues that i pointed out.

**Review by zeozeph — APPROVED** [review]
